editoast/src/generated_data/error/mod.rs

```diff
@@ -33,17 +77,72 @@ impl GeneratedData for ErrorLayer {
         let graph = Graph::load(infra_cache);
 
         // Generate the errors
-        track_sections::insert_errors(conn, infra, infra_cache, &graph)?;
-        signals::insert_errors(conn, infra, infra_cache)?;
-        speed_sections::insert_errors(conn, infra, infra_cache)?;
-        track_section_links::insert_errors(conn, infra, infra_cache)?;
-        switch_types::insert_errors(conn, infra, infra_cache)?;
-        switches::insert_errors(conn, infra, infra_cache)?;
-        detectors::insert_errors(conn, infra, infra_cache)?;
-        buffer_stops::insert_errors(conn, infra, infra_cache)?;
-        routes::insert_errors(conn, infra, infra_cache, &graph)?;
-        operational_points::insert_errors(conn, infra, infra_cache)?;
 
+        let infra_errors = generate_errors(
+            ObjectType::TrackSection,
+            infra_cache,
+            &graph,
+            &track_sections::TRACK_SECTIONS_ERRORS,
+        );
+        track_sections::insert_errors(infra_errors, conn, infra)?;
+        let infra_errors = generate_errors(
+            ObjectType::Signal,
+            infra_cache,
+            &graph,
+            &signals::SIGNALS_ERRORS,
+        );
+        signals::insert_errors(infra_errors, conn, infra)?;
+        let infra_errors = generate_errors(
+            ObjectType::SpeedSection,
+            infra_cache,
+            &graph,
+            &speed_sections::SPEED_SECTION_ERRORS,
+        );
+        speed_sections::insert_errors(infra_errors, conn, infra)?;
+
+        let infra_errors = generate_errors(
+            ObjectType::SwitchType,
+            infra_cache,
+            &graph,
+            &switch_types::SWITCH_TYPES_ERRORS,
+        );
+        switch_types::insert_errors(infra_errors, conn, infra)?;
+
+        let infra_errors = generate_errors(
+            ObjectType::Detector,
+            infra_cache,
+            &graph,
+            &detectors::DETECTOR_ERRORS,
+        );
+        detectors::insert_errors(infra_errors, conn, infra)?;
+        let infra_errors = generate_errors(
+            ObjectType::BufferStop,
+            infra_cache,
+            &graph,
+            &buffer_stops::BUFFER_STOP_ERRORS,
+        );
+        buffer_stops::insert_errors(infra_errors, conn, infra)?;
+
+        let infra_errors = generate_errors(
+            ObjectType::OperationalPoint,
+            infra_cache,
+            &graph,
+            &operational_points::OPERATIONAL_POINTS_ERRORS,
+        );
+        operational_points::insert_errors(infra_errors, conn, infra)?;
+
+        let infra_errors = generate_errors(
+            ObjectType::Route,
+            infra_cache,
+            &graph,
+            &routes::ROUTE_ERRORS,
+        );
+        routes::insert_errors(infra_errors, conn, infra)?;
+
+        // We can't use generate_errors function for switches, routes, track_section_links
+        // because we can't split the check functions using the same signature as for other objects
```

Comment:
Update this comment, this apply only to `switches` and `track_section_links` now.